WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateLogic.java

```diff
@@ -322,6 +329,24 @@ public void run() {
                 ReaderBlogList serverBlogs = ReaderBlogList.fromJson(jsonObject);
                 ReaderBlogList localBlogs = ReaderBlogTable.getFollowedBlogs();
 
+                // Remove duplicates from the server blogs list if local and remote doesn't match.
+                // This is required because under obscure circumstances the server can return duplicates.
+                // We could have modified the function isSameList to eliminate the length check,
+                // but it's better to keep it separate since we aim to remove this check as soon as possible.
+                if (serverBlogs.size() != localBlogs.size()) {
+                    for (int i = 0; i < serverBlogs.size(); i++) {
+                        ReaderBlog outer = serverBlogs.get(i);
+                        for (int j = serverBlogs.size() - 1; j > i; j--) {
+                            ReaderBlog inner = serverBlogs.get(j);
+                            if (outer.blogId == inner.blogId && outer.feedId == inner.feedId) {
```

Comment:
💡 I believe just checking the `blogId` is enough since that's the only primary key in our local database anyway, and based on the comments in `ReaderBlogTable`, `blogId` is the same as `feedId` for feeds, and `feedId` is always 0 for blogs.